express_relay/sdk/js/src/index.ts

```diff
@@ -218,33 +269,66 @@ export class Client {
     this.websocketOpportunityCallback = callback;
   }
 
+  public setBidStatusHandler(
+    callback: (statusUpdate: BidStatusUpdate) => Promise<void>
+  ) {
+    this.websocketBidStatusCallback = callback;
+  }
+
   /**
    * Subscribes to the specified chains
    *
    * The opportunity handler will be called for opportunities on the specified chains
    * If the opportunity handler is not set, an error will be thrown
    * @param chains
    */
-  async subscribeChains(chains: string[]) {
+  async subscribeChains(chains: string[]): Promise<void> {
     if (this.websocketOpportunityCallback === undefined) {
-      throw new Error("Opportunity handler not set");
+      throw new ClientError("Opportunity handler not set");
     }
-    return this.sendWebsocketMessage({
+    await this.sendWebsocketMessage({
       method: "subscribe",
       params: {
         chain_ids: chains,
       },
     });
   }
 
+  async submitOpportunityBidViaWebsocket(bid: OpportunityBid): Promise<BidId> {
+    const result = await this.sendWebsocketMessage({
+      method: "post_liquidation_bid",
+      params: {
+        opportunity_bid: this.toServerOpportunityBid(bid),
+        opportunity_id: bid.opportunityId,
+      },
+    });
+    if (result === null) {
+      throw new ClientError("Empty response in websocket for bid submission");
+    }
+    return result.id;
+  }
+
+  async submitBidViaWebsocket(bid: Bid): Promise<BidId> {
+    const result = await this.sendWebsocketMessage({
```

Comment:
nit: i think `sendWebsocketMessage` can be renamed to something that indicated there is a response but I can't think of a good name myself.